src/main/java/pedido/Pedido.java

```diff
@@ -27,23 +33,50 @@ public Cliente getCliente(){
     }
 
     public double calcularTotal(Cardapio cardapio){
-        double total= 0;
-        //TODO
+        double total = 0;
+
+        for (ItemPedido itemPedido : itens) {
+            total += itemPedido.getShake().getTipoTamanho().atualizaValorTamanho(cardapio.buscarPreco(itemPedido.getShake().getBase())) * itemPedido.getQuantidade();
+            for (Ingrediente adicional : itemPedido.getShake().getAdicionais()) {
+                total += cardapio.buscarPreco(adicional) * itemPedido.getQuantidade();
+            }
```

Comment:
Vale a pena procurar um pouco sobre map, reduce, que pode ajudar você a não utilizar o for nesse caso. Também acho legal você atribuir varíaveis na sua lógica, que ajuda a facilitar um pouco o entendimento.